### PR TITLE
June 21 sale

### PR DESCRIPTION
**Pull request by rupertbates**
## What are you doing in this PR?
We are going to run a promotional sale on newspaper subscriptions starting on the 5th of July. During this period we want all customers who purchase a print subscription to do so with the promo code `JUNE21SALE` so this PR adds this promo to the `DefaultPromotions` object and sets is as a default promotion on the paper landing page and the subscriptions landing page.

I have also taken the opportunity to rename several of the subs related controller classes to end in `...Controller` to be more consistent and to make it easier to tell from the class name what their purpose is.

[**Trello Card**](https://trello.com/c/lKZIn0DO/3725-print-campaign-subscriptions-landing-page-display-paper-offer)

### CODE REVIEW

File: support-frontend/app/controllers/PaperSubscriptionController.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ class PaperSubscription(
   def paper(): Action[AnyContent] = CachedAction() { implicit request =>
     implicit val settings: AllSettings = settingsProvider.getAllSettings()
     val canonicalLink = Some(buildCanonicalPaperSubscriptionLink())
-    val queryPromos = request.queryString.get("promoCode").map(_.toList).getOrElse(Nil)
+    val queryPromos = DefaultPromotions.Paper.june21Promotion :: request.queryString.get("promoCode").map(_.toList).getOrElse(Nil)
```
Comment (rupertbates) [inline support-frontend/app/controllers/PaperSubscriptionController.scala:41]: Add the default promotion to the subs page

File: support-frontend/app/controllers/SubscriptionsController.scala [hunk 2]
```diff
@@ -59,7 +59,7 @@ class Subscriptions(
     val service = priceSummaryServiceProvider.forUser(false)
 
     val paperMap = if (countryGroup == CountryGroup.UK) {
-      val paper = service.getPrices(Paper, Nil)(CountryGroup.UK)(Collection)(Sunday)(Monthly)(GBP)
+      val paper = service.getPrices(Paper, List(DefaultPromotions.Paper.june21Promotion))(CountryGroup.UK)(Collection)(Sunday)(Monthly)(GBP)
```
Comment (rupertbates) [inline support-frontend/app/controllers/SubscriptionsController.scala:62]: Add the default promotion to the subs landing page

File: support-frontend/app/controllers/PaperSubscriptionController.scala (hunk 1 above)
Comment (johnduffell) [reply support-frontend/app/controllers/PaperSubscriptionController.scala:41]: need to rename `queryPromos` as it's not just query string ones any more

**Review by johnduffell — APPROVED** [review]
sounds good to rename the controllers, I did get surprised by that before!
just a small comment but pretty simple and effective change. 👍

**Pull request merged**